File: go/all-your-base/all_your_base.go
package allyourbase

import (
	"errors"
	"math"
)

// ConvertToBase converts a number to the given base.
func ConvertToBase(inputBase int, digits []int, outputBase int) ([]int, error) {
	if inputBase < 2 {
		return []int{}, errors.New("input base must be >= 2")
	}
	if outputBase < 2 {
		return []int{}, errors.New("output base must be >= 2")
	}
	if len(digits) == 0 {
		return []int{0}, nil
	}
	res := []int{}
	tempNum := 0
	for i, exp := 0, len(digits)-1; i < len(digits); i, exp = i+1, exp-1 {
		if digits[i] < 0 || digits[i] >= inputBase {
			return []int{}, errors.New("all digits must satisfy 0 <= d < input base")
		}
		tempNum += digits[i] * int(math.Pow(float64(inputBase), float64(exp)))
	}
	if tempNum == 0 {
		return []int{0}, nil
	}
	for tempNum > 0 {
		res = append([]int{tempNum % outputBase}, res...)
		tempNum /= outputBase
	}

	return res, nil
}

File: kotlin/all-your-base/src/main/kotlin/AllYourBase.kt
class BaseConverter {
    var base10 = 0

    constructor(inBase: Int, digits: IntArray) {
        require(inBase >= 2) { "Bases must be at least 2." }
        require(digits.size >= 1) { "You must supply at least one digit." }
        require(digits.size == 1 || digits[0] != 0) { "Digits may not contain leading zeros." }

        var exp = (digits.size - 1).toDouble()

        for (digit in digits) {
            require(digit >= 0) { "Digits may not be negative." }
            require(digit < inBase) { "All digits must be strictly less than the base." }
            base10 += digit * Math.pow(inBase.toDouble(), exp--).toInt()
        }
    }

    fun convertToBase(newBase: Int): IntArray {
        require(newBase >= 2) { "Bases must be at least 2." }

        if (base10 == 0) {
            return intArrayOf(0)
        }

        var res = mutableListOf<Int>()

        while (base10 > 0) {
            res.add(0, base10 % newBase)
            base10 /= newBase
        }
        return res.toIntArray()
    }
}
